app/code/Magento/GoogleGtag/Block/Ga.php

```diff
@@ -145,23 +157,48 @@ public function getOrdersTrackingData(): array
         $collection = $this->orderRepository->getList($this->searchCriteriaBuilder->create());
 
         foreach ($collection->getItems() as $order) {
-            foreach ($order->getAllVisibleItems() as $item) {
-                $result['products'][] = [
-                    'item_id' => $this->escapeJsQuote($item->getSku()),
-                    'item_name' =>  $this->escapeJsQuote($item->getName()),
-                    'price' => number_format((float) $item->getPrice(), 2),
-                    'quantity' => (int)$item->getQtyOrdered(),
+            foreach ($order->getAllVisibleItems() as $index => $item) {
+                $product = $this->productRepository->get($item->getSku());
+                $orderProduct = [
+                    'index' => $index+1,
+                    'item_id' => $this->_escaper->escapeHtmlAttr($item->getSku()),
+                    'item_name' =>  $this->_escaper->escapeHtmlAttr($item->getName()),
+                    'item_brand' => $this->_escaper->escapeHtmlAttr(
+                        $product->getAttributeText('manufacturer')
+                    ),
+                    'affiliation' => $this->_escaper->escapeHtmlAttr(
+                        $this->_storeManager->getStore()->getFrontendName()
+                    ),
+                    'price' => round((float) $item->getPrice(), 2),
+                    'quantity' => (int)$item->getQtyOrdered()
                 ];
+
+                if ($item->getDiscountAmount() > 0) {
+                    $orderProduct['discount'] = $this->_escaper->escapeHtmlAttr($item->getDiscountAmount());
+
+                    if (!empty($order->getCouponCode())) {
+                        $orderProduct['coupon'] = $this->_escaper->escapeHtmlAttr($order->getCouponCode());
+                    }
+                }
+
+                $result['products'][] = $orderProduct;
             }
-            $result['orders'][] = [
+
+            $resultOrder = [
                 'transaction_id' =>  $order->getIncrementId(),
-                'affiliation' => $this->escapeJsQuote($this->_storeManager->getStore()->getFrontendName()),
-                'value' => number_format((float) $order->getGrandTotal(), 2),
-                'tax' => number_format((float) $order->getTaxAmount(), 2),
-                'shipping' => number_format((float) $order->getShippingAmount(), 2),
+                'currency' =>  $order->getOrderCurrencyCode(),
+                'value' => round($order->getGrandTotal(), 2),
```

Comment:
@Bashev here `round($order->getGrandTotal(), 2)` throws an error as round() expects int/float but it is getting string value here. You need to add a datatype float here as well. **'`value' => round((float) $order->getGrandTotal(), 2),`**

Reply:
@bhoopatparmar correct, thx.